packages/actor-query-operation-from/lib/ActorQueryOperationFrom.ts

```diff
@@ -0,0 +1,122 @@
+import {ActorQueryOperationTypedMediated, IActorQueryOperationOutput,
+  IActorQueryOperationTypedMediatedArgs} from "@comunica/bus-query-operation";
+import {IActorTest} from "@comunica/core";
+import * as RDF from "rdf-js";
+import {Algebra, Factory} from "sparqlalgebrajs";
+
+/**
+ * A comunica From Query Operation Actor.
+ */
+export class ActorQueryOperationFrom extends ActorQueryOperationTypedMediated<Algebra.From> {
+
+  private static FACTORY: Factory = new Factory();
+  private static ALGEBRA_TYPES: string[] = Object.keys(Algebra.types).map((key) => (<any> Algebra.types)[key]);
+
+  constructor(args: IActorQueryOperationTypedMediatedArgs) {
+    super(args, 'from');
+  }
+
+  /**
+   * Recursively transform the given operation to use the given graph as default graph
+   * This will create a new operation and not modify the given operation.
+   * @param {Operation} operation An operation.
+   * @param {RDF.Term} graph A graph term.
+   * @return {Operation} A new operation.
+   */
+  public static applyOperationGraph(operation: Algebra.Operation, graph: RDF.Term): Algebra.Operation {
+    // If the operation is a BGP or Path, change the graph.
+    if (operation.type === 'bgp') {
+      return ActorQueryOperationFrom.FACTORY.createBgp((<Algebra.Bgp> operation).patterns.map(
+        (quad) => quad.graph.termType === 'DefaultGraph'
+          ? ActorQueryOperationFrom.FACTORY.createPattern(quad.subject, quad.predicate, quad.object, graph) : quad));
+    } else if (operation.type === 'path') {
+      const path: Algebra.Path = <Algebra.Path> operation;
+      return ActorQueryOperationFrom.FACTORY.createPath(path.subject, path.predicate, path.object,
+        path.graph.termType === 'DefaultGraph' ? graph : path.graph);
+    }
+
+    // Otherwise, copy the operation and recursively call this method.
+    const copiedOperation: Algebra.Operation = <any> {};
+    for (const key of Object.keys(operation)) {
+      if (Array.isArray(operation[key])) {
+        if (key === 'variables') {
```

Comment:
I saw there is a similar private function in the sparql algebra lib. It would be nice to be able to reuse that. But that's not really urgent, perhaps something for a future cleanup phase :-)